File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& num, int target) {
        int n=num.size();
        int l=0,r=n-1;

        while(l<r){
            int x = num[l]+num[r];

            if(x==target) return {l+1, r+1};
            else if(x>target) r--;
            else l++;
        }
        return {-1, -1};
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-0.java
class Solution {
 public boolean isHappy(int num) {
        Set<Integer> st = new HashSet<>();

        int curr = num;

        while (curr != 1) {
            int n = curr;
            curr = 0;
            if (st.contains(n)) return false;
            while (n > 0) {
                int t = n % 10;
                n = n / 10;
                curr += t * t;
                // System.out.println(t);
            }
            st.add(curr);
            // System.out.println(curr);
        }
        return true;
    }
}
